### ISSUE TITLE
Invalid IPubsubMessage.data type

### ISSUE THREAD

**Issue by HNicolas** [googleapis/nodejs-pubsub#1150]
https://github.com/googleapis/nodejs-pubsub/blob/76db007f270a646e8570768fa827ea2a97b62cbc/protos/protos.d.ts#L441

`IPubsubMessage.data` type seems to be `Buffer` and it is typed as `UInt8Array|string`.

**Comment by feywind** [comment]
@HNicolas Thanks for the report!

@alexander-fenster This looks like an issue with proto generation maybe?

**Comment by alexander-fenster** [comment]
@feywind Yes, seems like I will need to add `Buffer` here: https://github.com/googleapis/gax-nodejs/blob/master/tools/compileProtos.ts#L141 - we do some serious hacking on the generated `protos.d.ts`. I will fix it (but probably not earlier than I come back from vacation), or feel free to make a fix there and release!  (it will, unfortunately, trigger regeneration of all libraries)

**Comment by feywind** [comment]
I'm moving this over to a cleanup item because it's going to cause a lot of churn, and maybe it's not the best time for it. I'm leaving it on the backlog here for now though.

**Comment by feywind** [comment]
I'm transferring this issue over to gax-nodejs since that's where the problem appears to be.
